pkg/graph/resolvers/plan_discussion.go

```diff
@@ -454,63 +454,73 @@ func CreateDiscussionReply(
 		if err != nil {
 			return reply, err
 		}
+
 		// Create Activity and notifications in the DB
 		_, notificationErr := notifications.ActivityTaggedInDiscussionReplyCreate(ctx, tx, principal.Account().ID, discussion.ID, reply.ID, reply.Content, loaders.UserNotificationPreferencesGetByUserID)
 		if notificationErr != nil {
 			return nil, fmt.Errorf("unable to generate notifications, %w", notificationErr)
 		}
-		go func() {
-
-			replyUser := principal.Account()
-			commonName := replyUser.CommonName
 
-			if err != nil {
-				logger.Error("error sending discussion reply emails. Unable to retrieve modelPlan",
-					zap.String("replyID", reply.ID.String()),
-					zap.Error(err))
-			}
+		pref, err := UserNotificationPreferencesGetByUserID(ctx, principal.Account().ID)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get user notification preference, Notification not created %w", err)
+		}
 
-			errReplyEmail := sendDiscussionReplyEmails(
-				ctx,
-				store,
-				logger,
-				emailService,
-				emailTemplateService,
-				addressBook,
-				discussion,
-				reply,
-				modelPlan,
-				replyUser,
-			)
-			if errReplyEmail != nil {
-				logger.Error("error sending tagged in plan discussion reply emails to tagged users and teams",
-					zap.String("discussionID", discussion.ID.String()),
-					zap.Error(errReplyEmail))
-			}
+		// Early return if
+		if !pref.NewDiscussionReply.SendEmail() {
```

Comment:
This is going to make some issues, the tag emails have their own preferences, so this specific preference shouldn't block those emails. (The tag emails are based on the preference of each individual who is tagged, while is this should be based on the preference of the user who created the discussion.